# Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
# Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not # Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not # Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not # Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not # Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not # Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not # Exception Handling in python

# Single Error in a try block
a = [1, 2, 3] 
try: 
  print ("Second element = %d" %(a[1]))
  
    # Throws error since there are only 3 elements in array 
  print ("Fourth element = %d" %(a[3]))  
  
except IndexError: 
  print ("An error occurred")

# Program to handle multiple errors with one except statement 
try :
  a = 7
  if a < 4 : 
  
    # throws ZeroDivisionError for a = 3  
    b = a/(a-3) 
      
    # throws NameError if a >= 4 
    print ("Value of b = ", b)
  
# note that braces () are necessary here for multiple exceptions 
except(ZeroDivisionError, NameError): 
  print ("\nError Occurred and Handled")

# Program to depict else clause with try-except 
  
# Function which returns a/b 
def AbyB(a , b): 
    try: 
      c = ((a+b) / (a-b)) 
    except ZeroDivisionError: 
        print ("a/b result in 0")
    else: 
        print (c)
  
# Driver program to test above function 
AbyB(2.0, 3.0) 
AbyB(3.0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not .0, 3.0) 

# Program to depict Raising Exception 
  
try:  
    raise NameError("Hi there")  # Raise Error 
except NameError: 
    print ("An exception")
    raise  # To determine whether the exception was raised or not 